lib/src/frontend/future_matchers.dart

```diff
@@ -80,3 +80,37 @@ class _Completes extends AsyncMatcher {
     return description;
   }
 }
+
+/// Matches a [Future] that does not complete.
+///
+/// Note that this creates an asynchronous expectation. The call to
+/// `expect()` that includes this will return immediately and execution will
+/// continue.
+final Matcher doesNotComplete = const _DoesNotComplete(20);
+
+class _DoesNotComplete extends Matcher {
+  final int _timesToPump;
+  const _DoesNotComplete(this._timesToPump);
+
+  // TODO(grouma) - Make this a top level function
+  Future _pumpEventQueue(times) {
+    if (times == 0) return new Future.value();
+    return new Future(() => _pumpEventQueue(times - 1));
+  }
+
+  Description describe(Description description) {
+    description.add("does not complete");
+    return description;
+  }
+
+  @override
+  bool matches(item, Map matchState) {
+    if (item is! Future) fail('$item is not a Future');
+    item.then((value) {
+      value = value;
+      fail('$item completed with a value of $value');
```

Comment:
Yeah I thought about that since it would mostly be <Instance of _Future> but opted to keep it just in case it was useful. I've removed it though.